_source/_posts/2019-05-30-okta-token-hooks.md

```diff
@@ -0,0 +1,395 @@
+
+For demonstration purposes, you're going to use the [ngrok]() service, which issues you a unique, publicly addressable domain that's connected to your locally running Spring Boot application.
+
+Start by issuing this command in a separate terminal tab:
+
+```bash
+ngrok http 8080
+```
+
+You'll see output like this:
+
+```bash
+ngrok by @inconshreveable                                                                                                                                                                     (Ctrl+C to quit)
+
+Session Status                online
+Session Expires               6 hours, 30 minutes
+Version                       2.3.29
+Region                        United States (us)
+Web Interface                 http://127.0.0.1:4040
+Forwarding                    http://e3fc9a95.ngrok.io -> http://localhost:8080
+Forwarding                    https://e3fc9a95.ngrok.io -> http://localhost:8080
+
+Connections                   ttl     opn     rt1     rt5     p50     p90
+                              0       0       0.00    0.00    0.00    0.00
+```
+
+Copy the https link (`https://e3fc9a95.ngrok.io` in my case) as we'll need it soon.
+
+### To the Bleeding Edge: Setup Okta Inline Hooks
+
+The publicly accessible (thanks to ngrok) Spring Boot app you have running is all setup to receive hook requests from Okta. But, your Okta org isn't ready for it yet.
+
+First, you need to enable the Early Access feature in your Okta org. To do that, Select **Developer Console** in the upper left and choose **Classic UI**.
+
+Click **Settings > Features** from the top level menu. Click **Edit**. Check **Inline Hooks**. Click **Save**. Now you can work with Inline Hooks in your Okta org.
+
+Next, you need to register the inline hook with Okta. There isn't yet a UI for this. You have to make an API call as documented [here]().
+
+However, built into the Spring Boot application is an API endpoint that makes the call to your Okta org on your behalf. It uses the API Token that you configured earlier to do this.
+
+It builds the request to Okta from a template file found in the project at: `src/main/resources/hooks.json`:
+
+```javascript
+{
+    "status": "ACTIVE",
+    "name": "APIAM Token Hook with DB lookup",
+    "type": "com.okta.oauth2.tokens.transform",
+    "version": "1.0.0",
+    "channel": {
+        "type": "HTTP",
+        "version": "1.0.0",
+        "config": {
+            "uri": "{{ngrok-url}}/api/hooks/apiam",
+            "headers": [{"key": "org", "value": "{{okta-org}}"}],
+            "authScheme": {
+                "type": "HEADER",
+                "key": "Authorization",
+                "value": "Basic {{basic-auth}}"
+            }
+        }
+    }
+}
+```
+
+The `{{ngrok-url}}` variable is the ngrok url you copied earlier. The `{{okta-org}}` variable is your Okta org url. These values are passed in as part of an api call you'll make shortly.
```

Comment:
Do these {{}} actually render? It's been a problem in the past needing {% raw %} {% endraw %} around them because Jeckyll uses these as replacement tokens.